internal/database/pebble_batch.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package database
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/pebble"
+)
+
+var _ Batch = (*pebbleBatch)(nil)
+
+type pebbleBatch struct {
+	batch *pebble.Batch
+}
+
+func (pb *pebbleBatch) Put(key, value []byte) error {
+	err := pb.batch.Set(key, value, &pebble.WriteOptions{})
```

Comment:
looks like this is ignored in `pebble.Batch.Set()`
```suggestion
	err := pb.batch.Set(key, value, nil)
```